neural_nets/activations/activations.py

```diff
@@ -149,3 +219,95 @@ def grad(self, x):
     def grad2(self, x):
         # 0 if x >= 0 else alpha * e^(z)
         return np.where(x >= 0, np.zeros_like(x), self.alpha * np.exp(x))
+
+
+class Exponential(ActivationBase):
+    """
+    Exponential (base e) activation function.
+    """
+
+    def __init__(self):
+        super().__init__()
+
+    def __str__(self):
+        return "Exponential"
+
+    def fn(self, z):
+        return np.exp(z)
+
+    def grad(self, x):
+        return np.exp(x)
+
+    def grad2(self, x):
+        return np.exp(x)
+
+
+class SELU(ActivationBase):
+    """
+    Scaled Exponential Linear Unit (SELU).
+
+    SELU is equal to: `scale * elu(x, alpha)`, where alpha and scale
+    are predefined constants. The values of `alpha` and `scale` are
+    chosen so that the mean and variance of the inputs are preserved
+    between two consecutive layers as long as the weights are initialized
+    correctly and the number of inputs is "large enough"
+    (see references for more information).
+
+    Note
+    ----------
+    - To be used together with the initialization "lecun_normal".
+    - To be used together with the dropout variant "AlphaDropout".
+
+    References
+    ----------
+    - [Self-Normalizing Neural Networks](https://arxiv.org/abs/1706.02515)
+    """
+
+    def __init__(self):
+        self.alpha = 1.6732632423543772848170429916717
+        self.scale = 1.0507009873554804934193349852946
+        self.elu = ELU(alpha=self.alpha)
+        super().__init__()
+
+    def __str__(self):
+        return "SELU"
+
+    def fn(self, z):
+        return self.scale * self.elu.fn(z)
+
+    def grad(self, x):
+        return np.where(x >= 0, np.ones_like(x) * self.scale, np.exp(x) * self.alpha * self.scale)
+
+    def grad2(self, x):
+        return np.where(x >= 0, np.zeros_like(x), np.exp(x) * self.alpha * self.scale)
+
+
+class HardSigmoid(ActivationBase):
+    """
+    Hard sigmoid activation function.
+
+    Faster to compute than sigmoid activation.
+
+    - `0` if `x < -2.5`
+    - `1` if `x > 2.5`
+    - `0.2 * x + 0.5` if `-2.5 <= x <= 2.5`.
+    """
+
+    def __init__(self):
+        super().__init__()
+
+    def __str__(self):
+        return "Hard Sigmoid"
+
+    def fn(self, z):
+        return np.clip((0.2 * z) + 0.5, 0.0, 1.0)
+
+    def grad(self, x):
+        arr = np.ones_like(x) * 0.2
+        for idx, num in enumerate(x):
+            if num < -2.5 or num > 2.5:
+                arr[idx] = 0.
+        return arr
```

Comment:
About `grad` function in `Hard Sigmoid`, if you have a better idea for coding this in numpy way, please tell me.

Reply:
You could do:
```python
def grad(self, x):
    # dx[i] = 0.2 if -2.5 <= x[i] <= 2.5 else 0
    return np.where((x >= -2.5) & (x <= 2.5), 0.2, 0)
```